### DISCUSSION TITLE
How to handle email address with + symbol in URI

### DISCUSSION THREAD

**Discussion by jonathanredford** [vercel/next.js#13885] [Help]
My app needs to access email address pass to it via the url query string.

In some case, a `+` will be present on the left side of the `@` however when accessing the query object either via the context in a server handler or `next/router` on the client side, the `+` is always replaced with a space.

Not sure if this is a bug or not so figured I'd post here before creating an issue to get some feedback from the community.

**Comment by oshanfernando** [comment]
'+' sign is decoded back to a space in a query string. So before you add the email to the query string, encode and then attach to preserve the special character.

Ex: "localhost:3000?email=" + encodeURIComponent('test+123@gmail.com')
